extension/src/main/java/synfron/reshaper/burp/core/rules/thens/ThenComment.java

```diff
@@ -19,8 +19,16 @@ public RuleResponse perform(EventInfo eventInfo) {
         boolean hasError = true;
         try {
             if (eventInfo.getAnnotations() != null) {
-                eventInfo.getAnnotations().setNotes(text.getText(eventInfo));
-                hasError = false;
+                if (!eventInfo.getAnnotations().notes().isEmpty()) {
```

Comment:
Thanks for the contribution!

I am concerned about changing the default functionality. A way to alleviate that is to add a boolean field (checkbox) labeled something like `Append` that would control whether or not append or overwrite.

That being said, I actually think this solution isn't needed. Today in the Text field, instead of putting `Hello World`, you can put `{{a:comment}}Hello World` instead. If there wasn't already a comment the final comment will be `Hello World`. However, if there is a comment already, it will be `<existing commment here>Hello World`. What are your thoughts on that?

Reply:
Hi @Arshiya0x7, were you able to try what I suggested above?

Reply:
Closing due to inactivity.